lms/djangoapps/instructor/utils.py

```diff
@@ -36,3 +39,120 @@ def get_module_for_student(student, usage_key, request=None):
     descriptor = modulestore().get_item(usage_key, depth=0)
     field_data_cache = FieldDataCache([descriptor], usage_key.course_key, student)
     return get_module(student, request, usage_key, field_data_cache)
+
+
+def collect_ora2_data(course_id):
+    """
+    Query MySQL database for aggregated ora2 response data.
+    """
+    cursor = get_read_replica_cursor_if_available(db)
+
+    #Syntax unsupported by other vendors such as SQLite test db
+    if db.connection.vendor != 'mysql':
+        return '', ['']
+
+    raw_queries = ora2_data_queries().split(';')
+
+    cursor.execute(raw_queries[0])
+    cursor.execute(raw_queries[1], [course_id])
+
+    header = [item[0] for item in cursor.description]
+
+    return header, cursor.fetchall()
+
+
+def ora2_data_queries():
+    """
+    Wraps a raw SQL query which retrieves all ORA2 responses for a course.
+    """
+
+    # pylint: disable=invalid-name
+    RAW_QUERY = """
```

Comment:
@sarina thanks for the tag.  I have some larger questions overall, and I'll comment at the bottom of this PR, but if we were to go ahead with this implementation I'd agree that in-lining the raw SQL makes sense, but I'd recommend making a common abstract class where this raw query is just one single implementation of how to get ORA2 data, and other implementations could use Django Models, or an entirely different query.

This still may run the risk of an injection, but only at a configuration level.